### ISSUE TITLE
Screen Readers Failing to Properly Interact with Tree Views

### ISSUE THREAD

**Issue by amerikrainian** [godotengine/godot#108958]
### Tested versions

- Reproducible in Godot 4.5 Beta 3

### System information

Godot v4.5.beta3 - Windows 11 (build 26100) - Multi-window, 1 monitor - Vulkan (Forward+) - dedicated NVIDIA RTX A1000 Laptop GPU (NVIDIA; 32.0.15.6641) - 12th Gen Intel(R) Core(TM) i7-12800H (20 threads) - 31.67 GiB memory. Using NVDA screen reader version 2025.1.2

### Issue description

Screen readers fail to properly parse tree views. This is seen most quickly with the file system tree view:
- When scrolling over the file system tree view for a new project, first item that gets announced: res://   data item  res://. There is no indication as to whether this can be expanded with the right arrow, but it can, in fact, be done. First press of the right arrow is silent, and then I get (after pressing right arrow one more time), "icon.svg   data item  icon.svg"
- Likewise, collapsing the tree view, e.g., left arrow on the res:// option, is also silent.
- It is also possible to scroll off this tree view by pressing down (or up) arrow(s) enough times. The latter brings you to the filter files text field, while the former takes you to some unlabeled separator (for me it reads, "   separator  270")

What should happen:
- Typically, screen readers report, when scrolling over, what could be collapsed, e.g., "res://  collapsed  1 of 1  level 0"
- If expanded, screen readers report this fact, e.g., "expanded" "1 item." From a brief test, this is actually 2 separate speech fragments as shown.
- Likewise, when collapsing the tree view item, they report, "collapsed"
- If scrolling over a non-collapsible item, screen readers report this, e.g, "icon.svg  1 of 1"
- It is impossible to scroll off the tree view unless one hits tab or shift tab explicitly (or, presumably, some other hotkey to snap them off to another part of the interface). Arrow keys can never take one out of the tree.

I am aware that was a lot of words. In lieu of this fact, I'm attaching an example of a better (though not ideal) behavior of Python's WX toolkit and screen readers, as I believe it's much easier to mess with it as opposed to parsing lots of jargon. It doesn't seem to report number of items within the tree properly, but it does report expanding, collapsing, and prevents you from arrowing to the close button.

```python
# Tested with wx 4.2.3, Python 3.13
import wx

class TreeViewDemo(wx.Frame):
    def __init__(self):
        super().__init__(None, title="Tree View Demo", size=(400, 500))
        
        panel = wx.Panel(self)
        
        sizer = wx.BoxSizer(wx.VERTICAL)
        
        self.tree = wx.TreeCtrl(panel, style=wx.TR_DEFAULT_STYLE | wx.TR_HAS_BUTTONS)
        
        root = self.tree.AddRoot("res://")
        
        self.tree.AppendItem(root, "icon.svg")
        self.tree.AppendItem(root, "project.godot")
        
        assets = self.tree.AppendItem(root, "assets/")
        self.tree.AppendItem(assets, "player.png")
        self.tree.AppendItem(assets, "background.jpg")
        
        scripts = self.tree.AppendItem(root, "scripts/")
        self.tree.AppendItem(scripts, "main.gd")
        self.tree.AppendItem(scripts, "player.gd")
        self.tree.AppendItem(scripts, "enemy.gd")
        
        scenes = self.tree.AppendItem(root, "scenes/")
        self.tree.AppendItem(scenes, "main.tscn")
        self.tree.AppendItem(scenes, "player.tscn")
        
        self.tree.Expand(root)
        
        close_btn = wx.Button(panel, wx.ID_CLOSE, "Close")
        close_btn.Bind(wx.EVT_BUTTON, lambda e: self.Close())
        
        sizer.Add(self.tree, 1, wx.EXPAND | wx.ALL, 5)
        sizer.Add(close_btn, 0, wx.ALIGN_CENTER | wx.ALL, 5)
        
        panel.SetSizer(sizer)
        self.Centre()

if __name__ == "__main__":
    app = wx.App()
    frame = TreeViewDemo()
    frame.Show()
    app.MainLoop()
```

### Steps to reproduce

- Create fresh Godot project
- Hit CTRL+Alt+P
- Tab over to the file system tree view
- Use the arrow keys and witness the bug(s)

### Minimal reproduction project (MRP)

N/A

**Comment by bruvzg** [comment]
It might be AccessKit limitation, we do set `expanded` state for the items as well as `EXPAND` + `COLLAPSE` actions, bet there's no obvious way to indicate if item is collapsable. 

Related https://github.com/AccessKit/accesskit/issues/32

**Comment by DataTriny** [comment]
I confirm that AccessKit does not yet support neither the expanded/collapsed state nor tree views.

**Comment by amerikrainian** [comment]
> It might be AccessKit limitation, we do set `expanded` state for the items as well as `EXPAND` + `COLLAPSE` actions, bet there's no obvious way to indicate if item is collapsable.
> 
> Related [AccessKit/accesskit#32](https://github.com/AccessKit/accesskit/issues/32)

I just added UIA tree view support to accesskit, so in theory we should no longer have this problem. I'll confess I'm unsure how to test this though.
